backend/app/controllers/spree/admin/promotion_codes_controller.rb

```diff
@@ -20,6 +20,24 @@ def index
           end
         end
       end
+
+      def new
+        @promotion = Spree::Promotion.accessible_by(current_ability, :read).find(params[:promotion_id])
+        @promotion_code = @promotion.promotion_codes.new
```

Comment:
just wondering why we are not using ActiveRecord `build` here. Is there a specific reason? I think you can even do `@promotion.build_promotion_code` right?

Same in the create method where a more conventional create. I'm taking a look at [scaffolded rails controller templates](https://github.com/rails/rails/blob/master/railties/lib/rails/generators/rails/scaffold_controller/templates/controller.rb.tt#L20).

Let me know!

Reply:
`@promotion.build_promotion_code` does not seems to exists : 
```
irb(main):009:0> promotion = Spree::Promotion.last
irb(main):010:0> promotion.build_promotion_code
Traceback (most recent call last):
        1: from (irb):10
NoMethodError (undefined method `build_promotion_code' for #<Spree::Promotion:0x00007f625b3b2e18>)
Did you mean?  build_promotion_category
```

And `new` or `build` return the same object : 
```
irb(main):011:0> promotion.promotion_codes.build
=> #<Spree::PromotionCode id: nil, promotion_id: 65, value: nil, created_at: nil, updated_at: nil, promotion_code_batch_id: nil>
irb(main):012:0> promotion.promotion_codes.new
=> #<Spree::PromotionCode id: nil, promotion_id: 65, value: nil, created_at: nil, updated_at: nil, promotion_code_batch_id: nil>
```

I've switched to `build`